docroot/modules/custom/layout_builder_custom/layout_builder_custom.module

```diff
@@ -775,6 +775,48 @@ function _layout_builder_custom_block_submit(array &$form, FormStateInterface $f
         }
 
         if ($block->getPluginId() === 'inline_block:uiowa_card') {
+          // Remove image or icon based on media_options selection.
+          $media_option = $form_state->getValue([
+            'settings',
+            'block_form',
+            'media_options',
+          ]);
+
+          if ($media_option) {
+
+            $form_image_selection = [
+              'settings',
+              'block_form',
+              'field_uiowa_card_image',
+              'selection',
+            ];
+
+            $form_icon_selection = [
+              'settings',
+              'block_form',
+              'field_uiowa_card_icon',
+              0,
+              'value',
+              'icon',
+            ];
+
+            if ($media_option === 'image') {
+              if ($form_state->hasValue($form_icon_selection)) {
+                $form_state->unsetValue($form_icon_selection);
+              }
+            }
+            if ($media_option === 'icon') {
```

Comment:
It seems like this should be an `elseif` since we would never want to have both an icon and an image.